### DISCUSSION TITLE
Picking up javax.ws.rs.core.Application sub class

### DISCUSSION THREAD

**Discussion by mimaom** [resteasy/resteasy#3878] [Q&A]
Hello

I am currently working on a framework using CDI extensions and need access to the sub class of the JAX RS javax.ws.rs.core.Application.
So I have a plain web application that has the main JAX-RS entry point like this:

```
@ApplicationPath("/api")
public class RestApplication extends Application {
}

```
And off course also some resources like this:

```
@Path("/business")
public class ApiBusinessService {
}

```
I would really like to be able to get access to RestApplication class (or actually the value of @ApplicationPath annotation - in this case the "api" value).

I had hoped to be able to pick this up inside my CDI extension, and I have also noticed the following code inside the RESTEasy class RestEasyCdiExtension:

```
public <T extends Application> void observeApplications(@Observes ProcessAnnotatedType<T> event, BeanManager beanManager)
{
...
}

```
This is exactly what I would like and I have tried adding this to my own framework CDI extension, but this method is never called.
I have added a break point inside the RestEasyCdiExtension in the above method to see if that is called, but it is not.

Am i missing something or is the RESTEasy framework picking this class up in another way which I could also use?

I am running inside JBoss EAP 7.4.5.GA (WildFly Core 15.0.13.Final-redhat-00001)

Thanks.

BR
Michael

**Comment by jamezp** [comment]
I just now realized I never replied to this. I'm sorry about that.

How is your framework being deployed? The `ResteasyCdiExtension` should be invoked assuming CDI is being used in the deployment. In newer versions of RESTEasy, targeting Jakarta REST 3.1 so it won't work in Jakarta EE 8, we've got https://github.com/resteasy/resteasy/blob/main/resteasy-cdi/src/main/java/org/jboss/resteasy/cdi/ContextProducers.java. You could use that as an example though of how to use a CDI producer.
